client/js/analysis.js

```diff
@@ -1,25 +1,110 @@
 window.onload = function() {
-  for (let i = 1; i < 6; i++) {
+  // Listen for submission click
+  const formSubmit = document.getElementById('sendbutton');
+  formSubmit.addEventListener('click', fetchResponse, false);
+};
+
+/**
+ * Fetches the data from the backend and populates the screen
+ *
+ */
+async function fetchResponse() {
+  const param = document.getElementById('searchbar').value;
+  const response = await fetch(`/api/analysis?name=${param}`);
+  const value = await response.json();
+  console.log(value);
+  populationHandler(value);
+}
+
+/**
+ * Handles populating all elements on the screen
+ *
+ * @param {object} json the json object
+ */
+function populationHandler(json) {
+  const charts = document.getElementById('charts');
+  const list = document.getElementById('list');
+
+  removeAllChildNodes(charts);
+  removeAllChildNodes(list);
+  populateDonutCharts(json);
+  const time = populateComments(json);
+  determineOverall(json, time*1500);
+}
+
+/**
+ * Puts the charts on the screen
+ *
+ * @param {object} json the json object
+ */
+function populateDonutCharts(json) {
+  const map = json.perspectiveMap;
+  for (const key in map) {
+    if (key) { // this is here bc of the linter
```

Comment:
that's not what the linter was trying to get you to do

the way that [`for...in`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) loops work is nuanced, so to ensure that it does what you are expecting it to do, you either need to check whether `key` is actually a property of `map` by checking if `map.hasOwnProperty(key)` is `true`:

```javascript
for (const key in map) {
  if (map.hasOwnProperty(key)) {
    addDonutChart(key, (100 - map[key] * 100).toString());
  }
}
```

or avoid using a `for...in` loop and use a `for...of` loop instead:

```javascript
for (const [key, value] of Object.entries(map)) {
  addDonutChart(key, (100 - value * 100).toString());
}
```